tools/clang/test/CodeGenHLSL/unroll/intact_child_loops.hlsl

```diff
@@ -0,0 +1,13 @@
+// RUN: %dxc -Od -E main -T ps_6_0 %s | FileCheck %s
+// CHECK: @main
+
+// Test case for when an unrolled loop has a child loop that's not unrolled.
```

Comment:
For traceability, you could add "regression test for GitHub #2006" here, and "Fixes #2006" to your pull request so that everything is linked together.